packages/babel-core/src/transformation/internal-plugins.ts

```diff
@@ -0,0 +1,135 @@
+import proposalPrivateMethods from "@babel/plugin-proposal-private-methods";
+import proposalPrivatePropertyInObject from "@babel/plugin-proposal-private-property-in-object";
+import proposalUnicodePropertyRegex from "@babel/plugin-proposal-unicode-property-regex";
+
+// We use functions here so that the `lazy` CJS transform avoids loading the
+// plugins until they are needed.
+const transformPlugins = {
+  "block-scoped-functions": () => transformBlockScopedFunctions,
+  "function-name": () => transformFunctionName,
+  // Merge these three plugins in a single reserved-words one?
+  "member-expression-literals": () => transformMemberExpressionLiterals,
+  "property-literals": () => transformPropertyLiterals,
+  "reserved-words": () => transformReservedWords,
+
+  // ES2015
+  "arrow-functions": () => transformArrowFunctions,
+  "block-scoping": () => transformBlockScoping, // RENAME: let-const
+  classes: () => transformClasses,
+  "computed-properties": () => transformComputedProperties,
+  destructuring: () => transformDestructuring,
+  "duplicate-keys": () => transformDuplicateKeys,
+  "for-of": () => transformForOf,
+  literals: () => transformLiterals,
+  "new-target": () => transformNewTarget,
+  "object-super": () => transformObjectSuper,
+  parameters: () => transformParameters,
+  regenerator: () => transformRegenerator, // RENAME: generators
+  spread: () => transformSpread, // RENAME: array-spread
+  "shorthand-properties": () => transformShorthandProperties,
+  "sticky-regex": () => transformStickyRegex, // RENAME: regex-sticky
+  "template-literals": () => transformTemplateLiterals,
+  "typeof-symbol": () => transformTypeofSymbol,
+  "unicode-regex": () => transformUnicodeRegex, // RENAME: regex-unicode
+  "unicode-escapes": () => transformUnicodeEscapes,
+
+  // ES2016
+  "exponentiation-operator": () => transformExponentiationOperator,
+
+  // ES2017
+  "async-to-generator": () => transformAsyncToGenerator, // RENAME: async-functions
+
+  // ES2018
+  "async-generator-functions": () => proposalAsyncGeneratorFunctions,
+  "dotall-regex": () => transformDotallRegex, // RENAME: regex-dotall
+  "named-capturing-groups-regex": () => transformNamedCapturingGroupsRegex, // RENAME: regex-...
+  "object-rest-spread": () => proposalObjectRestSpread,
+  "unicode-property-regex": () => proposalUnicodePropertyRegex, // RENAME: regex-...
+
+  // ES2019
+  "json-strings": () => proposalJsonStrings,
+  "optional-catch-binding": () => proposalOptionalCatchBinding,
+
+  // ES2020
+  "dynamic-import": () => proposalDynamicImport,
+  "export-namespace-from": () => proposalExportNamespaceFrom,
+  "nullish-coalescing-operator": () => proposalNullishCoalescingOperator,
+  "optional-chaining": () => proposalOptionalChaining,
+
+  // ES2021
+  "logical-assignment-operators": () => proposalLogicalAssignmentOperators,
+  "numeric-separator": () => proposalNumericSeparator, // RENAME: numeric-separators
```

Comment:
For most plugins after ES2021, they are named from the corresponding proposal slug:

https://github.com/tc39/proposal-numeric-separator
https://github.com/tc39/proposal-class-static-block

I believe it is straightforward if you want to try out a plugin after you learn that proposal. If TC39 can somehow enforce consistent naming among proposals, that would be great. Otherwise I suggest we rename it but keep old name as an alias.

Reply:
I can discuss it with other TC39 delegates.